### PR TITLE
Re-introduce fix for Hydra directory sync with multiple process

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Re-introduce fix from #5629 that was left out of the accelerator refactor highlighted by @mees! Also there seems there was an edge case that caused some issues with @mees setup, and he suggested a slight modification to the fix. 

Have confirmed locally that this fixes the issue.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Comment by SeanNaren** [comment]
I'll try get a test for this in place, I had difficulty the first time but this time round might be doable...

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 1]
```diff
@@ -159,6 +159,8 @@ def _call_children_scripts(self):
             if _HYDRA_AVAILABLE:
                 if HydraConfig.initialized():
                     cwd = get_original_cwd()
+                    os_cwd = f'"{os.getcwd()}"'
```
Comment (kaushikb11) [inline pytorch_lightning/plugins/training_type/ddp.py:162]:
```suggestion
                    os_cwd = os.getcwd()
```

Comment (mees) [reply pytorch_lightning/plugins/training_type/ddp.py:162]: Hey @kaushikb11, keeping ` os_cwd = f'"{os.getcwd()}"`' change is actually necesarry for cases where ${hydra.job.override_dirname} is used as part of the directory name, for example:
```
hydra:
  run:
    dir: ${log_dir}/runs/${now:%Y-%m-%d}/${now:%H-%M-%S}_${hydra.job.override_dirname}
```
without this change hydra thinks you are trying to override the run_dir with invalid characters that are used as separators (commas and = symbols). I know it doesn't look pretty, but I don't have a better solution right now.

Comment (SeanNaren) [reply pytorch_lightning/plugins/training_type/ddp.py:162]: This was purposefully set as if the user has set the directory to include some form of interpolation, we need to make sure this is evaluated as a string.

We should add a test case for this eventually.

**Review by kaushikb11 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**